cmd/mt-whisper-importer-reader/main.go

```diff
@@ -0,0 +1,350 @@
+	"gopkg.in/raintank/schema.v1"
+)
+
+var (
+	exitOnError = flag.Bool(
+		"exit-on-error",
+		true,
+		"Exit with a message when there's an error",
+	)
+	verbose = flag.Bool(
+		"verbose",
+		false,
+		"Write logs to terminal",
+	)
+	httpEndpoint = flag.String(
+		"http-endpoint",
+		"http://127.0.0.1:8080/chunks",
+		"The http endpoint to send the data to",
+	)
+	chunkSpanStr = flag.String(
+		"chunkspans",
+		"10min",
+		"List of chunk spans separated by ':'. The 1st whisper archive gets the 1st span, 2nd the 2nd, etc",
+	)
+	namePrefix = flag.String(
+		"name-prefix",
+		"",
+		"Prefix to prepend before every metric name, should include the '.' if necessary",
+	)
+	threads = flag.Int(
+		"threads",
+		10,
+		"Number of workers threads to process .wsp files",
+	)
+	skipUnfinishedChunks = flag.Bool(
+		"skip-unfinished-chunks",
+		true,
+		"Chunks that have not completed their chunk span should be skipped",
+	)
+	orgId = flag.Int(
+		"orgid",
+		1,
+		"Organization ID the data belongs to ",
+	)
+	whisperDirectory = flag.String(
+		"whisper-directory",
+		"/opt/graphite/storage/whisper",
+		"The directory that contains the whisper file structure",
+	)
+	readArchivesStr = flag.String(
+		"read-archives",
+		"*",
+		"Comma separated list of positive integers or '*' for all archives",
+	)
+	chunkSpans   []uint32
+	readArchives map[int]struct{}
+	printLock    sync.Mutex
+)
+
+func main() {
+	flag.Parse()
+
+	for _, chunkSpanStrSplit := range strings.Split(*chunkSpanStr, ",") {
+		chunkSpan := dur.MustParseUNsec("chunkspan", chunkSpanStrSplit)
+
+		if (mdata.Month_sec % chunkSpan) != 0 {
+			panic("chunkSpan must fit without remainders into month_sec (28*24*60*60)")
+		}
+
+		_, ok := chunk.RevChunkSpans[chunkSpan]
+		if !ok {
+			panic(fmt.Sprintf("chunkSpan %d is not a valid value (https://github.com/raintank/metrictank/blob/master/docs/memory-server.md#valid-chunk-spans)", chunkSpan))
+		}
+
+		chunkSpans = append(chunkSpans, chunkSpan)
+	}
+
+	if *readArchivesStr != "*" {
+		readArchives = make(map[int]struct{})
+		for _, archiveIdStr := range strings.Split(*readArchivesStr, ",") {
+			archiveId, err := strconv.Atoi(archiveIdStr)
+			if err != nil {
+				panic(fmt.Sprintf("Invalid archive id %q: %q", archiveIdStr, err))
+			}
+			readArchives[archiveId] = struct{}{}
+		}
+	}
+
+	fileChan := make(chan string)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(*threads)
+	for i := 0; i < *threads; i++ {
+		go processFromChan(fileChan, wg)
+	}
+
+	getFileListIntoChan(fileChan)
+	wg.Wait()
+}
+
+func throwError(msg string) {
+	msg = fmt.Sprintf("%s\n", msg)
+	if *exitOnError {
+		panic(msg)
+	} else {
+		printLock.Lock()
+		fmt.Fprintln(os.Stderr, msg)
+		printLock.Unlock()
+	}
+}
+
+func log(msg string) {
+	if *verbose {
+		printLock.Lock()
+		fmt.Println(msg)
+		printLock.Unlock()
+	}
+}
+
+func processFromChan(files chan string, wg *sync.WaitGroup) {
+	client := &http.Client{}
+
+	for {
+		file, more := <-files
+		if !more {
+			wg.Done()
+			return
+		}
```

Comment:
ah i was wrong. but even better, you can just do `for file := range files` and the loop will stop when files is closed so you can just put the `wg.Done()` at the end of this function